objects/LuaScript.ttslua

```diff
@@ -1628,8 +1634,249 @@ function setupMiscThings()
   end
 end
 
--- distribute starting player decks and tokens
+function setupCampaignMisc()
+  local campaignBag = getObjectFromGUID(bags.campaignItems)
+
+  -- take fear button
+  local buttonZone = getObjectFromGUID(deckZones.fear)
+  params = {
+    function_owner = self,
+    label = 'Take',
+    click_function = 'takeFearButton',
+    position = { 0, 0, -2.9 },
+    width = 800,
+    height = 380,
+    font_size = 200
+  }
+  buttonZone.createButton(params)
```

Comment:
Is this duplicate code? (EDIT: The stuff below too?

Reply:
Yes the fear buttons, the resource take buttons and the resource counter buttons code is identical to the non campaign mode.

Reply:
Alright then let's put those into a function and re-use it

Reply:
Broke out the function and modified the normal button function a bit to streamline it better with part of the code moving to the other function.